crates/ra_analysis/src/completion.rs

```diff
@@ -20,13 +21,13 @@ use crate::{
 
 pub use crate::completion::completion_item::{CompletionItem, InsertText, CompletionItemKind};
 
-/// Main entry point for copmletion. We run comletion as a two-phase process.
+/// Main entry point for completion. We run completion as a two-phase process.
 ///
 /// First, we look at the position and collect a so-called `CompletionContext.
 /// This is a somewhat messy process, because, during completion, syntax tree is
-/// incomplete and can look readlly weired.
+/// incomplete and can look really weird.
 ///
-/// Once the context is collected, we run a series of completion routines whihc
```

Comment:
One day, I'll bolt on a spell checker on top of rust-analyzer... Correct spell checking  really needs to know the language for the following features:

* distinguishing between definitions and references. Only defs should be spell checked. 
* handling escapes in string literals
* handling doc-syntax of comments